controllers/useraccount/console_usersettings.go

```diff
@@ -0,0 +1,205 @@
+// It first attempts to find the Rolebinding by name. If not found, it then looks for the Rolebinding by label.
+// If multiple Rolebindings are found by label, all of them are deleted.
+// The function returns an error if any occurred.
+func deleteRoleBinding(ctx context.Context, cl client.Client, userUID string) error {
+	name := ConsoleUserSettingsResourceNamePrefix + userUID
+	logger := log.FromContext(ctx)
+	logger.Info(fmt.Sprintf("deleting rolebinding with name %s", name))
+
+	var toDelete []client.Object
+	objectType := &v1.RoleBinding{TypeMeta: metav1.TypeMeta{Kind: "RoleBinding"}}
+
+	rb, err := getConsoleUserSettingObjectByName(ctx, cl, name, objectType)
+	if err != nil {
+		return err
+	}
+	if rb == nil {
+		// try with label
+		if rbList, err := getRoleBindingsByLabel(ctx, cl, userUID); err != nil || len(rbList) == 0 {
+			return err
+		} else if len(rbList) > 0 {
+			for key := range rbList {
+				toDelete = append(toDelete, &rbList[key])
+			}
+		}
+	} else {
+		toDelete = append(toDelete, rb)
+	}
+	if err := deleteResources(ctx, cl, toDelete); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getConfigMapByLabel(ctx context.Context, cl client.Client, userUID string) ([]corev1.ConfigMap, error) {
+	configMapList := &corev1.ConfigMapList{}
+	labels := map[string]string{ConsoleUserSettingsIdentifier: "true", ConsoleUserSettingsUID: userUID}
+	err := cl.List(ctx, configMapList, client.MatchingLabels(labels), client.InNamespace(UserSettingNS))
+	if err != nil {
+		return []corev1.ConfigMap{}, err
+	}
+	return configMapList.Items, nil
+}
+
+func getRolesByLabel(ctx context.Context, cl client.Client, userUID string) ([]v1.Role, error) {
+	roleList := &v1.RoleList{}
+	labels := map[string]string{ConsoleUserSettingsIdentifier: "true", ConsoleUserSettingsUID: userUID}
+	err := cl.List(ctx, roleList, client.MatchingLabels(labels), client.InNamespace(UserSettingNS))
+	if err != nil {
+		return []v1.Role{}, err
+	}
+	return roleList.Items, nil
+}
+
+func getRoleBindingsByLabel(ctx context.Context, cl client.Client, userUID string) ([]v1.RoleBinding, error) {
+	rbList := &v1.RoleBindingList{}
+	labels := map[string]string{ConsoleUserSettingsIdentifier: "true", ConsoleUserSettingsUID: userUID}
+	err := cl.List(ctx, rbList, client.MatchingLabels(labels), client.InNamespace(UserSettingNS))
+	if err != nil {
+		return []v1.RoleBinding{}, err
+	}
+	return rbList.Items, nil
+}
+
+// function getConsoleUserSettingObjectByName looks for resources of type ConfigMap, Role and Rolebinding in the namespace openshift-console-user-settings by name.
+// it returns obj,nil when the object is found by name
+// return nil, nil if the object is not found, i.e ignore the not found error
+// return nil, err if there is any other kind of error, or if the object type is not supported
+func getConsoleUserSettingObjectByName(ctx context.Context, cl client.Client, name string, object client.Object) (client.Object, error) {
```

Comment:
wouldn't it be simpler if we had 3 funcs, 1 per type of resource to find (cm, role, rolebinding), instead of this large `switch` block? For example: `getConsoleUserSettingConfigMap` + `getConsoleUserSettingRole` + `getConsoleUserSettingRoleBinding`?

Reply:
resolving this comment as this function is no longer present